### ISSUE TITLE
CSS modules are not correctly loaded

### ISSUE THREAD

**Issue by richardguerre** [yuanqing/create-figma-plugin#50]
**Problem**: when using `@create-figma-plugin/ui`, the CSS modules are not loaded correctly.

It appears to be because the css loader is not recognizing `.css` imports as CSS modules, instead it recognizes it as a regular css file, and the imported class/styles object is empty, so the resulting class names are empty:
![image](https://user-images.githubusercontent.com/53370763/129507522-e4c797d1-aada-4e8a-84ca-939922d33b0d.png)
Example of empty class names (the class has whitespace as the Button component has multiple classes concatenated together with spaces in between)

Context: I'm trying to set up Storybook with the new [Vite builder](https://github.com/eirslett/storybook-builder-vite). Also tried with the default Webpack builder, but I have similar issues. Even tried with a few Vite and Webpack plugins, like [vite-plugin-tranform-css-modules](https://github.com/Summer-andy/vite-plugin-transform-css-modules).

I tried patching the `@create-figma-plugin/ui` within `node_modules` directly by changing `button.css` to `button.module.css` and in `button.tsx` changed the `button.css` import to `button.module.css`. And that appears to fix the issue!
![image](https://user-images.githubusercontent.com/53370763/129511427-9b21cd54-73c6-4463-8451-e80bf25da12b.png)
![image](https://user-images.githubusercontent.com/53370763/129511474-ef80abea-d94e-4e42-9c5b-7b54df6149fd.png)

Any way we can change the CSS module `.css` imports to `.module.css`?
Or is there any config I need to change? like add some configs for `postcss` in the Vite config?

P.S. love the project!

**Comment by yuanqing** [comment]
@richardguerre Loading of CSS modules works fine for me both in the context of:

1. The plugin UI `iframe` in Figma itself (try `npx --yes -- create-figma-plugin my-plugin`, then select the `ui` template, then build, then run the plugin in Figma)
2. The [Storybook](https://yuanqing.github.io/create-figma-plugin/ui/) (the deployed version is identical to my local)

Below is the configuration for esbuild and Storybook to make `.css` files work:

https://github.com/yuanqing/create-figma-plugin/blob/ec7f4b79d8c120ece9d78f09f52107807bbc2bc0/packages/ui/.storybook/main.cjs#L10

https://github.com/yuanqing/create-figma-plugin/blob/ec7f4b79d8c120ece9d78f09f52107807bbc2bc0/packages/build/src/utilities/build-bundles-async/esbuild-css-modules-plugin.ts#L16

**Comment by richardguerre** [comment]
Thanks! That's helpful. I'll try and find a way to do something similar in the `viteFinal` config.

Also just to clarify, there is nothing wrong with `build-figma-plugin`, only the storybook setup.

**Comment by richardguerre** [comment]
@yuanqing I was able to fix the Webpack version of my Storybook by adding the same configs you did:
https://github.com/yuanqing/create-figma-plugin/blob/ec7f4b79d8c120ece9d78f09f52107807bbc2bc0/packages/ui/.storybook/main.cjs#L10

But I wasn't able to fix the Vite version of my Storybook, which I really wanted. So I ended copying everything inside `src/` of the `@create-figma-plugin/ui`, and renaming the CSS files, and that worked.

Just for reference for anyone else reading this issue, I tried the following:
- Using the exported `esbuildCssModulesPlugin(minify: boolean)` function from `create-figma-plugin/packages/build/src/utilities/build-bundles-async/esbuild-css-modules-plugin.ts` and adding it to the `viteFinal.esbuild.plugins` array in `.storybook/main.js`. But I ran into some import issues as `esbuild-css-modules-plugin.ts` appears to be a ES module, so `import` must be used instead of `require`, which doesn't seem possible inside `.storybook/main.js`. (see note below)
- Copying the contents of `esbuild-css-modules-plugin.ts` into my `.storybook/main.js` and changing, running into similar issues of not being able to import ES modules. I could have tried finding NPM packages that would work, but I had already spent too much time trying to debug this.

Note: I may just not be familiar enough with CommonJS vs ESM, so maybe someone more familiar might be able to debug it.

Feel free to close this issue.
